# README

## Table of Contents

| File Name | Description |
| -- | -- |
|[0_data.md](0_data.md) | Getting the raw data onto Atlas HPC|
|[1_align.md](1_align.md) | GSNAP alignment... in progress |
|[2\_DE\_analysis.md](2_DE_analysis.md) | Installing DESeq2 in R ... in progress |

## Timeline

Introduction of different technologies and their papers in chronological order. I'm probably missing a paper / technology somewhere, so feel free to send suggestions.

**1975** - Sanger Sequencing

* Sanger, F. and Coulson, A.R., 1975. [A rapid method for determining sequences in DNA by primed synthesis with DNA polymerase](https://www.semanticscholar.org/paper/A-rapid-method-for-determining-sequences-in-DNA-by-Sanger-Coulson/8ae9a57dbb5c0a7bbaa13f0ef3fde7a5cd027cd1). Journal of molecular biology, 94(3), pp.441-448.

**1996** - Usually cited as the first mention of nanopore for DNA and RNA seq

* Kasianowicz, J.J., Brandin, E., Branton, D. and Deamer, D.W., 1996. [Characterization of individual polynucleotide molecules using a membrane channel](https://www.semanticscholar.org/paper/Characterization-of-individual-polynucleotide-using-Kasianowicz-Brandin/4f05f0170b5ca6caabd351e2f6f2b1ec10d8a4a8). Proceedings of the National Academy of Sciences, 93(24), pp.13770-13773.

**2008** - First RNAseq paper according to the [Hitzeman et al, 2013](https://www.semanticscholar.org/paper/Genes%2C-behavior-and-next-generation-RNA-sequencing.-Hitzemann-Bottomly/9c05cc4538a272adc0006a2d3e1967b34ce6d1c8) review paper

* Nagalakshmi, U., Wang, Z., Waern, K., Shou, C., Raha, D., Gerstein, M. and Snyder, M., 2008. [The transcriptional landscape of the yeast genome defined by RNA sequencing.](https://www.semanticscholar.org/paper/The-Transcriptional-Landscape-of-the-Yeast-Genome-Nagalakshmi-Wang/447bf5edc72aa9b8c96f841ff11e8f47e89d2ec4) Science, 320(5881), pp.1344-1349.

**2008** - First Brain RNAseq paper according to the [Hitzeman et al, 2013](https://www.semanticscholar.org/paper/Genes%2C-behavior-and-next-generation-RNA-sequencing.-Hitzemann-Bottomly/9c05cc4538a272adc0006a2d3e1967b34ce6d1c8) review paper

* Mortazavi, A., Williams, B.A., McCue, K., Schaeffer, L. and Wold, B., 2008. [Mapping and quantifying mammalian transcriptomes by RNA-Seq](https://www.semanticscholar.org/paper/Mapping-and-quantifying-mammalian-transcriptomes-by-Mortazavi-Williams/ef117c95b92b68b751143155022a5c1a600afe5c). Nature methods, 5(7), pp.621-628.

**2008** - Introducing Illumina Sequencing

* Bentley, D.R., Balasubramanian, S., Swerdlow, H.P., Smith, G.P., Milton, J., Brown, C.G., Hall, K.P., Evers, D.J., Barnes, C.L., Bignell, H.R. and Boutell, J.M., 2008. [Accurate whole human genome sequencing using reversible terminator chemistry](https://www.semanticscholar.org/paper/Accurate-Whole-Human-Genome-Sequencing-using-Bentley-Balasubramanian/6dad16a6941b204c2f5f95d9cda6d0124d5a1a7b). nature, 456(7218), pp.53-59.

**2009** - Introducing PacBio single molecule real-time (SMRT)

* Eid, J., Fehr, A., Gray, J., Luong, K., Lyle, J., Otto, G., Peluso, P., Rank, D., Baybayan, P., Bettman, B. and Bibillo, A., 2009. [Real-time DNA sequencing from single polymerase molecules](https://www.semanticscholar.org/paper/Real-Time-DNA-Sequencing-from-Single-Polymerase-Eid-Fehr/b0588d6e0753e7e3f82c4aa4f1609764179b12de). Science, 323(5910), pp.133-138.

**2009** - Bowtie, updated to Bowtie2 in 2012 - type:aligner

* Langmead, B., Trapnell, C., Pop, M. and Salzberg, S.L., 2009. [Ultrafast and memory-efficient alignment of short DNA sequences to the human genome](https://www.semanticscholar.org/paper/Ultrafast-and-memory-efficient-alignment-of-short-Langmead-Trapnell/ebe875cf08dd398e0ed25f518502301c984a9afe). Genome biology, 10(3), pp.1-10.
* Langmead, B. and Salzberg, S.L., 2012. [Fast gapped-read alignment with Bowtie 2](https://www.semanticscholar.org/search?q=Fast%20gapped-read%20alignment%20with%20Bowtie%202&sort=relevance). Nature methods, 9(4), p.357.

* Bowtie2 GitHub - [https://github.com/BenLangmead/bowtie2](https://github.com/BenLangmead/bowtie2)

**2009** - TopHat, updated to TopHat2 in 2013 - type:aligner

* Trapnell, C., Pachter, L. and Salzberg, S.L., 2009. TopHat: discovering splice junctions with RNA-Seq. Bioinformatics, 25(9), pp.1105-1111.
* Kim, D., Pertea, G., Trapnell, C., Pimentel, H., Kelley, R. and Salzberg, S.L., 2013. [TopHat2: accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions](https://www.semanticscholar.org/paper/TopHat2%3A-accurate-alignment-of-transcriptomes-in-of-Kim-Pertea/5b9c6a2785995d4aa944006b8b5d579aa4a39d96). Genome biology, 14(4), pp.1-13.

* TopHat Website - [http://ccb.jhu.edu/software/tophat/index.shtml](http://ccb.jhu.edu/software/tophat/index.shtml)

**2009** - BWA - type:aligner

* Li, H. and Durbin, R., 2009. [Fast and accurate short read alignment with Burrows–Wheeler transform](https://www.semanticscholar.org/paper/Fast-and-accurate-short-read-alignment-with-Li-Durbin/b6de563c03eedf95d7e880a2aeb5688936ea1d26). bioinformatics, 25(14), pp.1754-1760.

**2010** - EdgeR, usually cited with their 2012 paper - type:DEanalysis

* Robinson, M.D., McCarthy, D.J. and Smyth, G.K., 2010. [edgeR: a Bioconductor package for differential expression analysis of digital gene expression data](https://www.semanticscholar.org/paper/edgeR%3A-a-Bioconductor-package-for-differential-of-Robinson-McCarthy/ec3d71a2fdd01968a6dc638ee261715a0f118c1e). Bioinformatics, 26(1), pp.139-140.
* McCarthy, D.J., Chen, Y. and Smyth, G.K., 2012. [Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation](https://www.semanticscholar.org/paper/Differential-expression-analysis-of-multifactor-to-McCarthy-Chen/571c3ea8cabd16ab0ae7a1a3495d3f3aca918e23). Nucleic acids research, 40(10), pp.4288-4297.

* EdgeR Bioconductor - [https://bioconductor.org/packages/release/bioc/html/edgeR.html](https://bioconductor.org/packages/release/bioc/html/edgeR.html)

**2010** - baySeq - type:DEanalysis

* Hardcastle, T.J. and Kelly, K.A., 2010. [baySeq: empirical Bayesian methods for identifying differential expression in sequence count data](https://www.semanticscholar.org/paper/baySeq%3A-Empirical-Bayesian-methods-for-identifying-Hardcastle-Kelly/35f03921a48e03687333b9d275bf5e58bd2520b8). BMC bioinformatics, 11(1), pp.1-14.

* baySeq Bioconductor - [https://bioconductor.org/packages/release/bioc/html/baySeq.html](https://bioconductor.org/packages/release/bioc/html/baySeq.html)

**2010** - GSNAP - type:aligner

* Wu, T.D. and Nacu, S., 2010. [Fast and SNP-tolerant detection of complex variants and splicing in short reads](https://www.semanticscholar.org/paper/Fast-and-SNP-tolerant-detection-of-complex-variants-Wu-Nacu/8b3f8643421c2cacef03fa2e9d8c1c81eed1ed0d). Bioinformatics, 26(7), pp.873-881.

**2012** - QuasiSeq - type:DEanalysis

* Lund, S.P., Nettleton, D., McCarthy, D.J. and Smyth, G.K., 2012. [Detecting differential expression in RNA-sequence data using quasi-likelihood with shrunken dispersion estimates](https://www.semanticscholar.org/paper/Detecting-Differential-Expression-in-RNA-sequence-Lund-Nettleton/701cae7fb41417989c384bcd8268a620c4669ca4). Statistical applications in genetics and molecular biology, 11(5).

* QuasiSeq CRAN - [https://cran.r-project.org/web/packages/QuasiSeq/index.html](https://cran.r-project.org/web/packages/QuasiSeq/index.html)

**2013** - STAR - type:aligner

* Dobin, A., Davis, C.A., Schlesinger, F., Drenkow, J., Zaleski, C., Jha, S., Batut, P., Chaisson, M. and Gingeras, T.R., 2013. [STAR: ultrafast universal RNA-seq aligner](https://www.semanticscholar.org/paper/STAR%3A-ultrafast-universal-RNA-seq-aligner-Dobin-Davis/78ce0c149860363bbbe34306c75a4454ad23828d). Bioinformatics, 29(1), pp.15-21.

* STAR GitHub - [https://github.com/alexdobin/STAR](https://github.com/alexdobin/STAR)

**2013** - MaSuRCA genome assembly

* Zimin, A.V., Marçais, G., Puiu, D., Roberts, M., Salzberg, S.L. and Yorke, J.A., 2013. [The MaSuRCA genome assembler](https://www.semanticscholar.org/paper/The-MaSuRCA-genome-assembler-Zimin-Mar%C3%A7ais/476273d8efa79c3067e55bc5b556e34c88225491). Bioinformatics, 29(21), pp.2669-2677.

**2013** - CUFFDIFF2 - type:DEanalysis

* Differential analysis of gene regulation at transcript resolution with RNA-seq

* CUFFDIFF website - [http://cole-trapnell-lab.github.io/cufflinks/getting_started/](http://cole-trapnell-lab.github.io/cufflinks/getting_started/)

**2014** - DESeq2 - type:DEanalysis

* Love, M.I., Huber, W. and Anders, S., 2014. [Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.](https://www.semanticscholar.org/paper/Moderated-estimation-of-fold-change-and-dispersion-Love-Huber/772f5fca88de0f6f38116d73cc32e23efe780a10) Genome biology, 15(12), pp.1-21.

  Hmm...`Love, M.I.` is also a coauthor on a recent RNASeq Review article:

  `Love, M.I.` (DESeq developer) and `Patro,R ` (Salmon developer) co-authored a paper together, and also a review. 

  * Love, M.I., Soneson, C. and Patro, R., 2018. [Swimming downstream: statistical analysis of differential transcript usage following Salmon quantification](https://www.semanticscholar.org/paper/Swimming-downstream%3A-statistical-analysis-of-usage-Love-Soneson/7395d792c67dcc33d84c00ac9bdc7cfce27b4625). F1000Research, 7.
  * Van den Berge, K., Hembach, K.M., Soneson, C., Tiberi, S., Clement, L., Love, M.I., Patro, R. and Robinson, M.D., 2019. [RNA sequencing data: hitchhiker's guide to expression analysis.](https://www.semanticscholar.org/paper/RNA-Sequencing-Data%3A-Hitchhiker's-Guide-to-Analysis-Berge-Hembach/54ffba1e7abd6305f0bd7fcd67d45c202330b25b)

* DESeq2 Bioconductor - [https://bioconductor.org/packages/release/bioc/html/DESeq2.html](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)

**2014** - featureCounts - type:count?

* Liao, Y., Smyth, G.K. and Shi, W., 2014. [featureCounts: an efficient general purpose program for assigning sequence reads to genomic features](https://www.semanticscholar.org/paper/featureCounts%3A-an-efficient-general-purpose-program-Liao-Smyth/cdbe8a265ef4bd8350722d2209fc6cc6290da1b3). Bioinformatics, 30(7), pp.923-930.

* featureCount Website - [http://subread.sourceforge.net/](http://subread.sourceforge.net/)

**2015** - NOISeq R - type:DEanalysis

* Tarazona, S., Furió-Tarí, P., Turrà, D., Pietro, A.D., Nueda, M.J., Ferrer, A. and Conesa, A., 2015. [Data quality aware analysis of differential expression in RNA-seq with NOISeq R/Bioc package](https://www.semanticscholar.org/paper/Data-quality-aware-analysis-of-differential-in-with-Tarazona-Furi%C3%B3-Tar%C3%AD/432c172e73949fac6bdd8788115140553ceff26a). Nucleic acids research, 43(21), pp.e140-e140.

* NOISeq Bioconductor - [https://bioconductor.org/packages/release/bioc/html/NOISeq.html](https://bioconductor.org/packages/release/bioc/html/NOISeq.html)

**2015** - HiSat... which was updated to HiSat2 in 2019? - type:aligner

* Kim, D., Langmead, B. and Salzberg, S.L., 2015. [HISAT: a fast spliced aligner with low memory requirements](https://www.semanticscholar.org/paper/HISAT%3A-a-fast-spliced-aligner-with-low-memory-Kim-Langmead/176490c7bbb20b5a64aca49d9dc3b75bdfd76d67). Nature methods, 12(4), pp.357-360.
* Kim, D., Paggi, J.M., Park, C., Bennett, C. and Salzberg, S.L., 2019. [Graph-based genome alignment and genotyping with HISAT2 and HISAT-genotype](https://www.semanticscholar.org/paper/Graph-based-genome-alignment-and-genotyping-with-Kim-Paggi/189b79cda928d58f695cf8323b9ce2196fc22409). Nature biotechnology, 37(8), pp.907-915.

* HiSat Website - [http://daehwankimlab.github.io/hisat2/](http://daehwankimlab.github.io/hisat2/)

**2016** - Kallisto - type:workflow

* Bray, N.L., Pimentel, H., Melsted, P. and Pachter, L., 2016. [Near-optimal probabilistic RNA-seq quantification](https://www.nature.com/articles/nbt.3519). Nature biotechnology, 34(5), pp.525-527.

* Kallisto Website - [https://pachterlab.github.io/kallisto/about](https://pachterlab.github.io/kallisto/about)

**2017** - Salmon, with discussion of methods in 2019/20ish - type:workflow

* Patro, R., Duggal, G., Love, M.I., Irizarry, R.A. and Kingsford, C., 2017. [Salmon provides fast and bias-aware quantification of transcript expression](https://www.semanticscholar.org/paper/Salmon-provides-fast-and-bias-aware-quantification-Patro-Duggal/fd2613c0e04eba3b8b068ffc0109c250387c2776). Nature methods, 14(4), pp.417-419.
* Srivastava, A., Malik, L., Sarkar, H., Zakeri, M., Almodaresi, F., Soneson, C., Love, M.I., Kingsford, C. and Patro, R., 2020. [Alignment and mapping methodology influence transcript abundance estimation](https://www.semanticscholar.org/paper/Alignment-and-mapping-methodology-influence-Srivastava-Malik/8dfb98f6befa35e24d11331cd3d543ec0e9bd73a). Genome biology, 21(1), pp.1-29.

  Hmm... seems to be some strange discussion on Salmon vs Kallisto
  
  * [https://liorpachter.wordpress.com/2017/08/02/how-not-to-perform-a-differential-expression-analysis-or-science/](https://liorpachter.wordpress.com/2017/08/02/how-not-to-perform-a-differential-expression-analysis-or-science/)

* Salmon Website - [https://combine-lab.github.io/salmon/](https://combine-lab.github.io/salmon/)

**2018** - Ah...Hitzemann has a recent review paper:

* Hitzemann, R., Darakjian, P., Oberbeck, D., Zheng, C., Walter, N., Iancu, O.D., Searles, R., Harrington, C.A. and McWeeney, S., 2018. [Genes, Behavior, and Next-Generation Sequencing: The First 10 Years](https://www.semanticscholar.org/paper/Genes%2C-behavior%2C-and-next-generation-sequencing%3A-10-Hitzemann-Darakjian/2da210112b07a00d1eb7bece39536e60cb9f8260). In Molecular-Genetic and Statistical Techniques for Behavioral and Neural Research (pp. 289-308). Academic Press.

## Brainstorming questions

### What are the earliest aligner and DE analysis programs?

Maybe Bowtie and TopHat (earliest papers at 2009), and both were updated in 2012.

### What are the most recently developed aligner and DE analysis programs?

Maybe HiSat2 (2019), Salmon (2017), and NOISeq R (2015)? The Salmon and DESeq people seem to be working together.
