R Analysis
================
Jennifer Chang
1/8/2021

## R Markdown

This is a placeholder to store any analysis in R (edgeR, DESeq2). Also a
way to look at the underlying assumptions (RNASeq distribution functions
and how it behaves in an analysis pipeline. Is the distribution
realistic?)

## Install any required libraries

``` r
# == Install any CRAN packages
# cran_pkgs <- c("QuasiSeq")
# install.packages(cran_pkgs)

# == Install any Bioconductor packages
# bioc_pkgs <- c("edgeR", "baySeq", "DESeq2", "ballgown", "NOISeq", "limma")
# BiocManager::install(bioc_pkgs)

# ====================   Year of development
library(tidyverse)
library(magrittr)
library(edgeR)         # 2010
library(baySeq)        # 2010
library(QuasiSeq)      # 2012
library(DESeq2)        # 2014    # <= actually a bit strange that DESeq2 & edgeR tend to be equivalent, despite being 4yrs later
library(ballgown)      # 2014
library(NOISeq)        # 2015
library(limma)         # 2015
```

<!--  ## For now comment out exploration portion.
## Distribution

RNASeq is assumed to be [negatively binomially distributed](https://en.wikipedia.org/wiki/Negative_binomial_distribution) according to the edgeR and DESeq2 papers. 

$$
Y_{gi} \sim NB(M_{i}p{gj}, \phi_{g})
$$

What does a negative binomial distribution look like?


```r
y  <- rnbinom( n = 1000,       # Randomly generate 1000 entries (number of replicates?)
               size = 5,       # at least 5 successes (expressed/not?)
               prob = 0.5      # probability of success in each case...maybe length of genome may affect this
               )

df <- data.frame(x = c(1:1000),
                 y = y)
                  
ggplot(data = df, aes(x=y)) + 
  geom_histogram(binwidth = 1) +
  theme_bw() +
  labs(title="Negative Binomial Distribution (n=1000, size=5, prob=0.5)") +
  geom_vline(xintercept=5, color="blue")
```

![](imgs/2_DE_analysisnegbinomial-1.png)<!-- -->

Notice how the peak of the distribution is around 5 (`size=5`). Play
around with the `rnbinom` parameters, how does that change the shape of
the distribution? What implications does this have for RNASeq to
determine if a gene is differentially expressed across two treatments?
–&gt;

# Load in RNAseq counts

Load in the RNAseq counts generated by GSNAP.

``` r
data_maize <- read_delim("maize_genecounts.txt", delim = "\t")
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   .default = col_double(),
#>   Geneid = col_character()
#> )
#> ℹ Use `spec()` for the full column specifications.
data_bee <- read_delim("bee_genecounts.txt", delim="\t")
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   .default = col_double(),
#>   Geneid = col_character()
#> )
#> ℹ Use `spec()` for the full column specifications.
```

# Diagnostic plots

These are quick plots to look at the data and notice if any of the
counts seem unusally high or low, or multimodal. It can indicate a
problem with a sample, but does not necessarily mean there is a problem
with a sample.

## Bee

``` r
melt_bee <- data_bee %>%
  pivot_longer(!Geneid, names_to = "replicate", values_to = "count") %>%
  mutate(
    group=substr(replicate, 1, 3)                 # <= replace this with metadata information later
  )

melt_bee %>% 
  ggplot(., aes(x=replicate, y=count, group=Geneid)) + 
  geom_line(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5)
  ) +
  labs(title="GSNAP RNAseq Gene Profiles")
```

![](2_DE_analysisbee_linechart-1.png)<!-- -->

Distribution, violin plot

``` r
melt_bee %>% 
  ggplot(., aes(x=replicate,y=log(count))) +       # <= log transformed as a rough normalization... we will use DESeq2's normalization eventually
  geom_violin(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90)
  ) +
  labs(title="GSNAP distribution of gene counts by sample") +
  facet_wrap(~group, scales="free", drop=T)
#> Warning: Removed 216873 rows containing non-finite values (stat_ydensity).
```

![](2_DE_analysisviolin-1.png)<!-- -->

Check if any of the replicates seem obviously different from the others.
Mostly seems fine, `1-E07-F3-S59 distribution` is a little low compared
to others, `1-B03-A16_S21` seems a little high compared to others.

## Maize

``` r
melt_maize <- data_maize %>%
  pivot_longer(!Geneid, names_to = "replicate", values_to = "count")

melt_maize %>% 
  ggplot(., aes(x=replicate, y=count, group=Geneid)) + 
  geom_line(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5)
  ) +
  labs(title="GSNAP RNAseq Gene Profiles")
```

![](2_DE_analysismaize_linechart-1.png)<!-- -->

``` r
melt_maize %>% 
  ggplot(., aes(x=replicate,y=log(count))) +       # <= log transformed as a rough normalization... we will use DESeq2's normalization eventually
  geom_violin(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90)
  ) +
  labs(title="GSNAP distribution of gene counts by sample")
#> Warning: Removed 473515 rows containing non-finite values (stat_ydensity).
```

![](2_DE_analysismaize_violin-1.png)<!-- -->

``` r
#  facet_wrap(~group, scales="free", drop=T)
```

Check if any of the replicates seem obviously different from the others.
Mostly seems okay, `SRR1573509` and `SRR1573523` is a little low
compared to others, `SRR1573513` seems a little high compared to others.

## DESEQ2

Most of these commands I got from Siva, working on the WGCNA tutorial
and other notes from Siva.

``` r
# library(DESeq2)

de_input = as.matrix(data_bee[-1])
row.names(de_input) = data_bee$Geneid
de_input[1:5,1:6]
#>                   1-A01-A1_S7 1-A02-A2_S8 1-A03-A3_S9 1-A04-A4_S10 1-A05-A5_S11
#> gene-LOC100740276          14           3          17           11            8
#> gene-LOC100740157         100         108         103           75           74
#> gene-LOC100742884         186         117         159          127          125
#> gene-LOC100740399          25          19          12            5           11
#> gene-LOC100740519         139          90          74           79           86
#>                   1-A06-A6_S12
#> gene-LOC100740276           21
#> gene-LOC100740157          102
#> gene-LOC100742884          120
#> gene-LOC100740399           20
#> gene-LOC100740519          118

metadata <- readr::read_delim("../bumblebee_meta.csv", delim=",")
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   Well_ID = col_character(),
#>   Sample_Name = col_character(),
#>   Trt = col_character(),
#>   Nest = col_double()
#> )
row.names(metadata) = metadata$Well_ID
#> Warning: Setting row names on a tibble is deprecated.

meta_df <- data.frame( Sample = colnames(de_input)) %>%
  separate(Sample, 
           c(NA, "group", NA, NA), 
           sep="-", 
           remove=FALSE) %>%
  mutate(
    Condition = metadata[group,]$Trt,
    Condition = factor(Condition, levels=c("ctrl", "exposed"))
  )
#> Warning: Expected 4 pieces. Missing pieces filled with `NA` in 60 rows [1, 2, 3,
#> 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].
```

Start DESeq2, modified from Masonbrink’s notes.

``` r
dds <- DESeqDataSetFromMatrix(
  de_input,                # Gene counts
  colData = meta_df,       # Metadata
  design = ~Condition      # experimental design
)
#> converting counts to integer mode
dds <- DESeq(dds)
#> estimating size factors
#> estimating dispersions
#> gene-wise dispersion estimates
#> mean-dispersion relationship
#> final dispersion estimates
#> fitting model and testing
#> -- replacing outliers and refitting for 33 genes
#> -- DESeq argument 'minReplicatesForReplace' = 7 
#> -- original counts are preserved in counts(dds)
#> estimating dispersions
#> fitting model and testing
res <- results(dds)
table(res$padj<0.05)
#> 
#> FALSE  TRUE 
#>  7947   335
res <- res[order(res$padj), ]
resdata <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by="row.names", sort=FALSE)
names(resdata)[1] <- "Gene"
write.csv(resdata, file="AllExposedvsAllControlGene.csv",quote = FALSE,row.names = F)

rld <- rlogTransformation(dds)
#> rlog() may take a long time with 50 or more samples,
#> vst() is a much faster transformation
summary(rld)
#> [1] "DESeqTransform object of length 12925 with 24 metadata columns"

(p <- plotPCA(rld, intgroup="Condition") +
    theme_bw() +
    labs(x = "PC1", y="PC2")
)
```

![](2_DE_analysispca_bee-1.png)<!-- -->

## Other diagnostic plots

Rough outline

``` r
ordered_trt = c(meta_df %>% subset(Condition=="ctrl") %>% {.$Sample},
                meta_df %>% subset(Condition=="exposed") %>% {.$Sample})

DEgenes <- resdata %>%
  subset(padj < 0.05) %>%
  select(-c(2:7)) %>%
  pivot_longer(col=-1, names_to="treatment", values_to="expression") %>% 
  separate(treatment, 
           c(NA, "group", NA, NA), 
           sep="-", 
           remove=FALSE) %>%
  mutate(
    Condition = metadata[group,]$Trt,
    Condition = factor(Condition, levels=c("ctrl", "exposed")),
    treatment = factor(treatment, levels=ordered_trt)
  )
#> Warning: Expected 4 pieces. Missing pieces filled with `NA` in 20100 rows [1, 2,
#> 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].

# Very rough draft, fix this later...
DEgenes %>% ggplot(., aes(x=treatment, y=expression, group=Gene)) +
  geom_line(alpha=0.5) +
  geom_point(aes(color=Condition), size=0.5) +
  theme_bw() + 
  theme(
    axis.text.x = element_text(angle=90)
  )# +
```

![](2_DE_analysisbee_deline-1.png)<!-- -->

``` r
#  facet_wrap(~Condition, scales="free_x", drop=T)
```

Hmm… plot the top 20 differentially expressed genes (lowest padj
values)… rough draft, go back and check on normalization and metadata…

``` r
top20 = resdata[c(1:20),] %>%
  subset(padj < 0.05) %>%
  select(-c(2:7)) %>%
  pivot_longer(col=-1, names_to="treatment", values_to="expression") %>% 
  separate(treatment, 
           c(NA, "group", NA, NA), 
           sep="-", 
           remove=FALSE) %>%
  mutate(
    Condition = metadata[group,]$Trt,
    Condition = factor(Condition, levels=c("ctrl", "exposed")),
    treatment = factor(treatment, levels=ordered_trt)
  )
#> Warning: Expected 4 pieces. Missing pieces filled with `NA` in 1200 rows [1, 2,
#> 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].

top20 %>% ggplot(., aes(x = as.factor(Gene), y = expression, fill=Condition))+
  geom_boxplot() + 
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=45, hjust = 1)
  ) +
  labs(x="Gene")
```

![](2_DE_analysisbee_top20-1.png)<!-- -->
