
Week of	Topic	Reading ideas (open to suggestion)!
**27-Jan**	
History of sequencing and analysis 	Hitzemann, R., Bottomly, D., Darakjian, P., Walter, N., Iancu, O., Searles, R., Wilmot, B. and McWeeney, S., 2013. Genes, behavior and next‐generation RNA sequencing. Genes, Brain and Behavior, 12(1), pp.1-12.

Kukurba, K.R. and Montgomery, S.B., 2015. RNA sequencing and analysis. Cold Spring Harbor Protocols, 2015(11), pp.pdb-top084970.

<span style="color:Blue">**3-Feb**</span>	
Costa-Silva, J., Domingues, D. and Lopes, F.M., 2017. RNA-Seq differential expression analysis: An extended review and a software tool. PloS one, 12(12), p.e0190152.

Raplee, I.D., Evsikov, A.V. and Marín de Evsikova, C., 2019. Aligning the Aligners: Comparison of RNA sequencing data alignment and gene expression quantification tools for clinical breast cancer research. Journal of personalized medicine, 9(2), p.18.

Alshehri, H. and Alkharouf, N., 2018, December. Compare and contrast of differential gene expression software packages of RNA-Seq. In 2018 International Conference on Computational Science and Computational Intelligence (CSCI) (pp. 1374-1379). IEEE.


<span style="color:Green">**10-Feb**</span>	Networks part I	Iancu, O.D., Kawane, S., Bottomly, D., Searles, R., Hitzemann, R. and McWeeney, S., 2012. Utilizing RNA-Seq data for de novo coexpression network inference. Bioinformatics, 28(12), pp.1592-1597.

Sinha, S., Jones, B.M., Traniello, I.M., Bukhari, S.A., Halfon, M.S., Hofmann, H.A., Huang, S., Katz, P.S., Keagy, J., Lynch, V.J. and Sokolowski, M.B., 2020. Behavior-related gene regulatory networks: A new level of organization in the brain. Proceedings of the National Academy of Sciences, 117(38), pp.23270-23279.


15-Feb	Networks part II	Patalano, S., Vlasova, A., Wyatt, C., Ewels, P., Camara, F., Ferreira, P.G., Asher, C.L., Jurkowski, T.P., Segonds-Pichon, A., Bachman, M. and González-Navarrete, I., 2015. Molecular signatures of plastic phenotypes in two eusocial insect species with simple societies. Proceedings of the National Academy of Sciences, 112(45), pp.13970-13975.

22-Feb	Gene modules part I	Stone, E.A. and Ayroles, J.F., 2009. Modulated modularity clustering as an exploratory tool for functional genomic inference. PLoS Genet, 5(5), p.e1000479.
1-Mar	Gene modules part II	Morandin, C., Brendel, V.P., Sundström, L., Helanterä, H. and Mikheyev, A.S., 2019. Changes in gene DNA methylation and expression networks accompany caste specialization and age‐related physiological changes in a social insect. Molecular ecology, 28(8), pp.1975-1993.

8-Mar	Machine learning part I	Jabeen, A., Ahmad, N. and Raza, K., 2018. Machine learning-based state-of-the-art methods for the classification of rna-seq data. In Classification in BioApps (pp. 133-172). Springer, Cham.

15-Mar	Machine learning part II	Wyatt, C.D., Bentley, M., Taylor, D., Brock, R.E., Taylor, B.A., Bell, E., Leadbeater, E. and Sumner, S., 2020. Genetic toolkit for sociality predicts castes across the spectrum of social complexity in wasps. bioRxiv.

22-Mar	Comparative analysis part I: across methods	Jaakkola, M.K., Seyednasrollah, F., Mehmood, A. and Elo, L.L., 2017. Comparison of methods to detect differentially expressed genes between single-cell populations. Briefings in bioinformatics, 18(5), pp.735-743.

Soneson, C. and Delorenzi, M., 2013. A comparison of methods for differential expression analysis of RNA-seq data. BMC bioinformatics, 14(1), p.91.


29-Mar	Comparative analysis part II: across species	Hunt, V.L., Hino, A., Yoshida, A. and Kikuchi, T., 2018. Comparative transcriptomics gives insights into the evolution of parasitism in Strongyloides nematodes at the genus, subclade and species level. Scientific reports, 8(1), pp.1-9.

Cheng, H., Wang, Y. and Sun, M.A., 2018. Comparison of gene expression profiles in nonmodel eukaryotic organisms with RNA-seq. In Transcriptome Data Analysis (pp. 3-16). Humana Press, New York, NY.

Wolf, T., Kämmer, P., Brunke, S. and Linde, J., 2018. Two's company: studying interspecies relationships with dual RNA-seq. Current opinion in microbiology, 42, pp.7-12.

5-Apr	Computational workgroup meeting- work on final project
12-Apr	Computational workgroup meeting- work on final project
19-Apr	Work on final project AND/OR begin final presentations (depending on number of participants)
26-Apr	Finish final presentations

*Topics denoted as "part I" will consist mostly of background to the analysis technique and a comparison to conventional pipelines; "part II" will mostly consist of examples of the technique in the literature and a discussion about what the technique does well (or poorly) compared to conventional techniques.
**Final projects will consist at least one practical application of the gene expression analyses covered in class. You are encouraged to use your own data set. Presentations should be approx. 15 minutes long
