---
title: "R Analysis"
author: "Jennifer and Mou"
date: "1/8/2021"
output:
  github_document
editor_options:
  chunck_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  collapse = TRUE,
  comment = "#>",
  fig.path = "imgs/2_DE_analysis"
)
```

## R Markdown

This is a placeholder to store any analysis in R (edgeR, DESeq2). Also a way to look at the underlying assumptions (RNASeq distribution functions and how it behaves in an analysis pipeline. Is the distribution realistic?)

## Install any required libraries

```{r, warning=FALSE, message=FALSE}
# == Install any CRAN packages
# cran_pkgs <- c("QuasiSeq")
# install.packages(cran_pkgs)

# == Install any Bioconductor packages
# bioc_pkgs <- c("edgeR", "baySeq", "DESeq2", "ballgown", "NOISeq", "limma")
# BiocManager::install(bioc_pkgs)

# ====================   Year of development
library(tidyverse)
library(magrittr)
#library(edgeR)         # 2010
#library(baySeq)        # 2010
#library(QuasiSeq)      # 2012
library(DESeq2)        # 2014    # <= actually a bit strange that DESeq2 & edgeR tend to be equivalent, despite being 4yrs later
#library(ballgown)      # 2014
#library(NOISeq)        # 2015
#library(limma)         # 2015
```



# Load in RNAseq counts

Load in the RNAseq counts generated by GSNAP.

```{r}
data_maize <- read_delim("~/Desktop/maize/maize.genecounts.out.txt", delim = "\t")
data_bee <- read_delim("~/Desktop/bee/bee.genecounts.out.txt", delim="\t")
```

# Diagnostic plots

These are quick plots to look at the data and notice if any of the counts seem unusally high or low, or multimodal. It can indicate a problem with a sample, but does not necessarily mean there is a problem with a sample.

## Bee

```{r bee_linechart, fig.width=8, fig.height=5}
melt_bee <- data_bee %>%
  pivot_longer(!Geneid, names_to = "replicate", values_to = "count")  %>%
  mutate(
    group=substr(replicate, 1, 3)                 # <= replace this with metadata information later (A1-A19, B1-B*)
  )

melt_bee %>% 
  ggplot(., aes(x=replicate, y=count, group=Geneid)) + 
  geom_line(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5)
  ) +
  labs(title="GSNAP RNAseq Bee Gene Profiles") #try boxplot with group?
```

Distribution, violin plot

```{r violin, fig.width=11, fig.height=8}
melt_bee %>% 
  ggplot(., aes(x=replicate,y=log(count))) +       # <= log transformed as a rough normalization... we will use DESeq2's normalization eventually
  geom_violin(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90)
  ) +
  labs(title="GSNAP distribution of bee gene counts by sample") +
  facet_wrap(~group, scales="free", drop=T)
```

Check if any of the replicates seem obviously different from the others. Mostly seems fine, `1-E07-F5-S61` (exposed) is a little low compared to others, missing `1-B03-A16_S21` (control).

## Maize

```{r maize_linechart, fig.width=8, fig.height=5,eval=FALSE}
melt_maize <- data_maize %>%
  pivot_longer(!Geneid, names_to = "replicate", values_to = "count")

melt_maize %>% 
  ggplot(., aes(x=replicate, y=count, group=Geneid)) + 
  geom_line(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5)
  ) +
  labs(title="GSNAP RNAseq Maize Gene Profiles")
```

```{r maize_violin, fig.width=8, fig.height=4, eval=FALSE}
melt_maize %>% 
  ggplot(., aes(x=replicate,y=log(count))) +       # <= log transformed as a rough normalization... we will use DESeq2's normalization eventually
  geom_violin(alpha=0.5) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=90)
  ) +
  labs(title="GSNAP distribution of maize gene counts by sample")
#  facet_wrap(~group, scales="free", drop=T)
```

Check if any of the replicates seem obviously different from the others. Mostly seems okay, `SRR1573523` and `SRR1573525` is a little low compared to others, `SRR1573513` seems a little high compared to others.

## DESEQ2

Most of these commands Jennifer got from Siva, working on the WGCNA tutorial and other notes from Siva.

```{r}
# library(DESeq2)

# bee

de_input = as.matrix(data_bee[-1])
row.names(de_input) = data_bee$Geneid
de_input[1:5,1:6]

metadata <- readr::read_delim("~/Desktop/2021_workshop_transcriptomics/bumblebee_meta.csv", delim=",")
row.names(metadata) = metadata$Well_ID

meta_df <- data.frame( Sample = colnames(de_input)) %>%
  separate(Sample, 
           c(NA, "group", NA, NA), #NA doesn't create column
           sep="-", 
           remove=FALSE) %>%
  mutate(
    Condition = metadata[group,]$Trt,
    Condition = factor(Condition, levels=c("ctrl", "exposed"))) # factor help save memory, string saved as numbers. Deseq2 will take first factor as "wildtype"); can tell ggplot what order to list groups in legend
# ignore missing pieces
# metadata["A01",]$Trt <= gives you "ctrl"
# metadata["B07",]$Trt <= gives you "exposed"

# Played around with mutate function, seeing what happens when I change metadata[group,]$Trt to metadata[Well_ID,]$Nest. 
# Then decided to merge test2 and metadata dataframes to get Trt and Nest columns in test2 dataframe 
test <- data.frame( Sample = colnames(de_input)) %>%
  separate(Sample, 
           c(NA, NA, "SampleName"), #NA doesn't create column
           sep="-", 
           remove=FALSE) 
test2 <- test %>% separate(SampleName,
                  c("Sample_Name", NA, NA, NA, NA),
                  sep = "_",
                  remove = FALSE)
test3 <- merge(metadata, test2, by = "Sample_Name")
test4 <- test3[-c(6)] #drop SampleName column


# maize

de_inputm = as.matrix(data_maize[-1])
row.names(de_inputm) = data_maize$Geneid
de_inputm[1:5,1:6]

# Need metadata for maize!
metadatam <- readr::read_delim("~/Desktop/maize/maize_meta.csv", delim=",")
row.names(metadatam) = metadatam$run_accession

#remove all columns except for sample_title, run_accession

meta_dfm <- data.frame( Sample = colnames(de_inputm)) %>%
  separate(Sample, 
           c(NA, "group", NA, NA), #NA doesn't create column
           sep="-", 
           remove=FALSE) %>%
  mutate(
    Condition = metadatam[group,]$Trt,
    Condition = factor(Condition, levels=c("ctrl", "exposed"))) # factor help save memory, string saved as numbers. Deseq2 will take first factor as "wildtype"); can tell ggplot what order to list groups in legend
# ignore missing pieces
# metadata["A01",]$Trt <= gives you "ctrl"
# metadata["B07",]$Trt <= gives you "exposed"

# Played around with mutate function, seeing what happens when I change metadata[group,]$Trt to metadata[Well_ID,]$Nest. 
# Then decided to merge test2 and metadata dataframes to get Trt and Nest columns in test2 dataframe 
test <- data.frame( Sample = colnames(de_input)) %>%
  separate(Sample, 
           c(NA, NA, "SampleName"), #NA doesn't create column
           sep="-", 
           remove=FALSE) 
test2 <- test %>% separate(SampleName,
                  c("Sample_Name", NA, NA, NA, NA),
                  sep = "_",
                  remove = FALSE)
test3 <- merge(metadata, test2, by = "Sample_Name")
test4 <- test3[-c(6)] #drop SampleName column
```

Start DESeq2, modified from Masonbrink's notes.

```{r pca_bee, fig.width=5, fig.height=3}
# bee
dds <- DESeqDataSetFromMatrix(
  de_input,                # Gene counts
  colData = meta_df,       # Metadata
  design = ~Condition      # experimental design
) #set up specialized object
dds <- DESeq(dds) #actually running deseq
res <- results(dds)
table(res$padj<0.05) #looking at which ones are sig. diff. from control at 95% CI
#TRUE = 217 genes are diff. expressed
#FALSE = 8292
res <- res[order(res$padj), ] #come back to this later (take it apart). Order orders it in ascending order
resdata <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by="row.names", sort=FALSE)
names(resdata)[1] <- "Gene"
write.csv(resdata, file="AllExposedvsAllControlGene.csv",quote = FALSE,row.names = F)

rld <- rlogTransformation(dds) #vst() much faster transformation ... check it out
summary(rld)
# "DESeqTransform object of length 12925 with 24 metadata columns"
(p <- plotPCA(rld, intgroup="Condition") +
    theme_bw() +
    labs(x = "PC1", y="PC2")
)

# with test4 file - it works! Got same results as dds above
dds <- DESeqDataSetFromMatrix(
  de_input,                # Gene counts
  colData = test4,       # Metadata
  design = ~Trt      # experimental design
) #set up specialized object
dds <- DESeq(dds) #actually running deseq
res <- results(dds)
table(res$padj<0.05) #looking at which ones are sig. diff. from control at 95% CI
#TRUE = 217 genes are diff. expressed
#FALSE = 8292
res <- res[order(res$padj), ] #come back to this later (take it apart). Order orders it in ascending order
resdata <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by="row.names", sort=FALSE)
names(resdata)[1] <- "Gene"
write.csv(resdata, file="Bee_AllExposedvsAllControlGene.csv",quote = FALSE,row.names = F)

rld <- rlogTransformation(dds) #vst() much faster transformation ... check it out
summary(rld)
# "DESeqTransform object of length 12925 with 24 metadata columns"
(p <- plotPCA(rld, intgroup="Trt") +
    theme_bw() +
    labs(x = "PC1", y="PC2")
)
```

## Other diagnostic plots

PCA by Nest + treatment
ANOVA stats to look at treatment, nest effects on variation (only genes with large fold-change, p<0.05)

Rough outline

```{r bee_deline, fig.width=8, fig.height=4}
# bee
ordered_trt = c(meta_df %>% subset(Condition=="ctrl") %>% {.$Sample},
                meta_df %>% subset(Condition=="exposed") %>% {.$Sample})

DEgenes <- resdata %>%
  subset(padj < 0.05) %>%
  select(-c(2:7)) %>%
  pivot_longer(col=-1, names_to="treatment", values_to="expression") %>% 
  separate(treatment, 
           c(NA, "group", NA, NA), 
           sep="-", 
           remove=FALSE) %>%
  mutate(
    Condition = metadata[group,]$Trt,
    Condition = factor(Condition, levels=c("ctrl", "exposed")),
    treatment = factor(treatment, levels=ordered_trt)
  )

# Very rough draft, fix this later...
DEgenes %>% ggplot(., aes(x=treatment, y=expression, group=Gene)) +
  geom_line(alpha=0.5) +
  geom_point(aes(color=Condition), size=0.5) +
  theme_bw() + 
  theme(
    axis.text.x = element_text(angle=90)
  )# +
#  facet_wrap(~Condition, scales="free_x", drop=T)
```
Hmm... plot the top 20 differentially expressed genes (lowest padj values)... rough draft, go back and check on normalization and metadata...

```{r bee_top20, fig.width=8, fig.height=4}
# bee
top20 = resdata[c(1:20),] %>%
  subset(padj < 0.05) %>%
  select(-c(2:7)) %>%
  pivot_longer(col=-1, names_to="treatment", values_to="expression") %>% 
  separate(treatment, 
           c(NA, "group", NA, NA), 
           sep="-", 
           remove=FALSE) %>%
  mutate(
    Condition = metadata[group,]$Trt,
    Condition = factor(Condition, levels=c("ctrl", "exposed")),
    treatment = factor(treatment, levels=ordered_trt)
  )

top20 %>% ggplot(., aes(x = as.factor(Gene), y = expression, fill=Condition))+
  geom_boxplot() + 
  theme_bw() +
  theme(
    axis.text.x = element_text(angle=45, hjust = 1)
  ) +
  labs(x="Gene")
```
