# Check the quality of the reads and process

1. Log onto Nova HPC

```sh
ssh kmcelroy@nova.its.iastate.edu
```

2. Make directories to work in

```sh
cd /work/LAS/serb-lab/kmcelroy/2021_workshop_transcriptomics
mkdir 03_qc_maize
mkdir 04_qc_bee
```

3. Run FastQC on the Maize data to assess the quality of the reads. I ran this job array using the same list from 00_rawdata.md

```sh
!/bin/bash

# Copy/paste this job script into a text file and submit with the command:
#    sbatch thefilename

#SBATCH --time=4:00:00   # walltime limit (HH:MM:SS)
#SBATCH --nodes=1   # number of nodes
#SBATCH --ntasks-per-node=1   # 1 processor core(s) per node
#SBATCH --job-name="fastqc"
#SBATCH --mail-user=kmcelroy@iastate.edu   # email address
#SBATCH --mail-type=END
#SBATCH --mail-type=FAIL
#SBATCH --output="fastqc" # job standard output file (%j replaced by job id)
#SBATCH --array=0-23 # run 24 jobs in an array

# LOAD MODULES, INSERT CODE, AND RUN YOUR PROGRAMS HERE

module load fastqc/0.11.7-3flwcvl

cd /work/LAS/serb-lab/kmcelroy/2021_workshop_transcriptomics
names=($(cat sra))
echo ${names[${SLURM_ARRAY_TASK_ID}]}

fastqc -o "03_qc_maize/${names[${SLURM_ARRAY_TASK_ID}]}" "01_data_maize/${names[${SLURM_ARRAY_TASK_ID}]}_1.fastq.gz" "01_data_maize/${names[${SLURM_ARRAY_TASK_ID}]}_2.fastq.gz"
```

Run the above script:

```sh
cd /work/LAS/serb-lab/kmcelroy/2021_workshop_transcriptomics
sbatch fastqc_maize.sbatch
```

4. Use multiqc to summarize all of the FastQC results
```sh
module load py-multiqc/1.5-py2-lqqx3ht
cd 03_qc_maize
multiqc .
```

I uploaded the multiqc aggregation of the Maize FastQC reports to the folder QC. The data overall looks quite good but could use some processing.

The following reads, for example, failed the mean quality score assessment (honestly, they still look good though ... score drops a bit at the end).

```
SRX699507_1
SRX699507_2
SRX699508_2
SRX699509_1
SRX699509_2
SRX699510_1
SRX699510_2
SRX699513_1
SRX699513_2
```

5. Use fastp to trim adapters and low quality bases and filter out low quality reads.

I downloaded the fastp (v 0.21.0) binary from the Github repo for fastp [https://github.com/OpenGene/fastp]
```
# pwd = /home/kmcelroy
wget http://opengene.org/fastp/fastp
chmod a+x ./fastp
# I moved fastp to my binaries directory, a permanent location on my PATH
mv fastp bin/
```

Run fastp on the Maize data to assess the quality of the reads. I ran this job array using the same list of SRA accession IDs

```
#!/bin/bash

# Copy/paste this job script into a text file and submit with the command:
#    sbatch thefilename

#SBATCH --time=4:00:00   # walltime limit (HH:MM:SS)
#SBATCH --nodes=1   # number of nodes
#SBATCH --ntasks-per-node=1   # 1 processor core(s) per node
#SBATCH --job-name="fastp"
#SBATCH --mail-user=kmcelroy@iastate.edu   # email address
#SBATCH --mail-type=END
#SBATCH --mail-type=FAIL
#SBATCH --output="fastp" # job standard output file (%j replaced by job id)
#SBATCH --array=0-23 # run 24 jobs in an array

# LOAD MODULES, INSERT CODE, AND RUN YOUR PROGRAMS HERE

cd /work/LAS/serb-lab/kmcelroy/2021_workshop_transcriptomics
names=($(cat sra))
echo ${names[${SLURM_ARRAY_TASK_ID}]}

fastp -i "01_data_maize/${names[${SLURM_ARRAY_TASK_ID}]}_R1.fastq.gz" -I "01_data_maize/${names[${SLURM_ARRAY_TASK_ID}]}_R2.fastq.gz" -o "03_qc_maize/fastp/${names[${SLURM_ARRAY_TASK_ID}]}_R1.trim.fastq.gz" -O "03_qc_maize/fastp/${names[${SLURM_ARRAY_TASK_ID}]}_R2.trim.fastq.gz" -h "03_qc_maize/fastp/${names[${SLURM_ARRAY_TASK_ID}]}.html" -j "03_qc_maize/fastp/${names[${SLURM_ARRAY_TASK_ID}]}.json"
```

Run the above script:

```sh
cd /work/LAS/serb-lab/kmcelroy/2021_workshop_transcriptomics
sbatch fastp_maize.sbatch
```
