# Notes, Ideas and Thoughts

## To do
1. Bee
* Run fastqc and multiQC on fastq files
* Run multiQC on `*txt.summary` files
* Run combine.R
* I noticed we already have mapping sam files for bee. Run through samtools and compare featurecounts of this with your featurecount summaries.

2. Maize
* Run multiQC on fastq files
* Reran gsnap + featurecounts (job 154052) - how does output look?, run multiQC on `*txt.summary` files. Run combine.R

3. Read MultiQC to assess read alignment (can try this on gsnap output): http://www.bea.ki.se/documents/Intro2RNAseq.pdf

4. What parameter to use in gsnap to remove bad reads? (unassigned: unmapped, ambiguous, multimapping, no features)

<details><summary>blah</summary>
dfdfd
</details>
